### DISCUSSION TITLE
App Update Management

### DISCUSSION THREAD

**Discussion by ToS0** [caprover/caprover#2016] [Q&A] [answered]
Hi, I have a question on how to maintain pre-defined one-click-apps.

For example, I use Nextcloud (NC) as defined in [nextcloud.yml](https://github.com/caprover/one-click-apps/blob/master/public/v4/apps/nextcloud.yml) since the early days of CapRover (CR). The standard setup contains of 4 containers and I update some manually by providing my own Dockerfile via CR GUI, e.g. refering to the latest NC image like `FROM nextcloud:28.0.2-apache`

I am not sure whether this the suggested approach to keep container current, or what are best practices here. For example, what should be done with old installations when you update your YML from above. Re-setup and keeping the data?

Specifically I ran into a problem with the Cron container. The CapRover app YML refers to a [docker-compose-yml](https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/with-nginx-proxy/mariadb-cron-redis/apache/docker-compose.yml) which does not exist anymore. Now I have no idea how to keep the container updated.

Thanks a lot for your support!

**Comment by githubsaturn** [comment, marked as answer]
There is no single solution that works for everything, see here: https://caprover.com/docs/one-click-apps.html#upgrading-one-click-apps

**Reply by ToS0** [reply]
Got it, thanks!
